File: javascript/hamming/hamming.js

var Hamming = function() {};

Hamming.prototype.compute = function(input1, input2) {
    if (input1.length != input2.length) {
        throw new Error('DNA strands must be of equal length.')
    }

    var hammingDistance = 0
    
    for (var i = 0; i < input1.length; i++) {
        if (input1[i] !== input2[i]) {
            hammingDistance++
        }
    }

    return hammingDistance
};

module.exports = Hamming;

File: python/hamming/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")

    result = 0;
    for i in range(0, len(strand_a)):
        if (strand_a[i] != strand_b[i]):
            result += 1

    return result